src/loki/worker.cc

```diff
@@ -323,16 +319,22 @@ loki_worker_t::work(const std::list<zmq::message_t>& job,
         break;
       default:
         // apparently you wanted something that we figured we'd support but havent written yet
-        return jsonify_error({107}, info, request);
+        throw valhalla_exception_t{107};
     }
-    return result;
   } catch (const valhalla_exception_t& e) {
     LOG_WARN("400::" + std::string(e.what()) + " request_id=" + std::to_string(info.id));
-    return jsonify_error(e, info, request);
+    result = jsonify_error(e, info, request);
   } catch (const std::exception& e) {
     LOG_ERROR("400::" + std::string(e.what()) + " request_id=" + std::to_string(info.id));
-    return jsonify_error({199, std::string(e.what())}, info, request);
+    result = jsonify_error({199, std::string(e.what())}, info, request);
   }
+
+  // keep track of the metrics
+  if (!result.intermediate)
+    enqueue_statistics(request);
```

Comment:
...make this call to suck any statistics out to the queue to be pushed to statsd on `cleanup`. note that we only do this when the result isnt intermediate (which means that it will be returned to the caller at this stage rather than go on to a deeper stage of the pipeline)